{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fdc85a5f",
   "metadata": {},
   "source": [
    "## Import Packages and Set Directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f427b0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import argparse\n",
    "import os\n",
    "os.chdir('/Users/abc6435/Desktop/RABvcfs')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e32e88ce",
   "metadata": {},
   "source": [
    "## Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10a8cf6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import Population File\n",
    "pop_file=\"/Users/abc6435/Desktop/RABvcfs/pops.txt\"\n",
    "pops = pd.read_csv(pop_file, sep=\"\\t\", header=None)\n",
    "popA = pops[pops[0]==\"popA\"][1].astype(str)\n",
    "popB = pops[pops[0]==\"popB\"][1].astype(str)\n",
    "\n",
    "#Import VCF\n",
    "vcf_file=\"/Users/abc6435/Desktop/RABvcfs/rab_chr29_test.vcf\"\n",
    "with open(vcf_file) as file:\n",
    "    for line in file:\n",
    "        if line.startswith('#CHROM'):\n",
    "            cols = line.lstrip(\"#\").strip().split(\"\\t\")\n",
    "data = pd.read_csv(vcf_file, sep='\\t', comment='#', names=cols)\n",
    "\n",
    "#Extract Genotype Information\n",
    "loci = data[['CHROM','POS','REF','ALT']]\n",
    "samples = data.drop(columns=['CHROM', 'POS', 'REF', 'ALT','ID','QUAL','FILTER','INFO','FORMAT'])\n",
    "samples = samples.apply(lambda col: col.str.split(\":\").str[0])\n",
    "alleles = pd.concat([loci,samples], axis=1)\n",
    "\n",
    "#Count Alleles\n",
    "aA = alleles[popA].astype(str)\n",
    "alleles['popA_0'] = aA.apply(lambda col: col.str.count('0')).sum(axis=1) \n",
    "alleles['popA_1'] = aA.apply(lambda col: col.str.count('1')).sum(axis=1)\n",
    "aB = alleles[popB].astype(str)\n",
    "alleles['popB_0'] = aB.apply(lambda col: col.str.count('0')).sum(axis=1)\n",
    "alleles['popB_1'] = aB.apply(lambda col: col.str.count('1')).sum(axis=1)\n",
    "\n",
    "#Simulated Data Boolean\n",
    "simulated_vcf = True\n",
    "\n",
    "#Calculate Frequencies (Empirical VCF)\n",
    "alleles['f_popA_0'] = alleles['popA_0']/(alleles['popA_0'] + alleles['popA_1'])\n",
    "alleles['f_popA_1'] = alleles['popA_1']/(alleles['popA_0'] + alleles['popA_1'])\n",
    "alleles['f_popB_0'] = alleles['popB_0']/(alleles['popB_0'] + alleles['popB_1'])\n",
    "alleles['f_popB_1'] = alleles['popB_1']/(alleles['popB_0'] + alleles['popB_1'])\n",
    "alleles['nA'] = (alleles['popA_0'] + alleles['popA_1']) / 2\n",
    "alleles['nB'] = (alleles['popB_0'] + alleles['popB_1']) / 2\n",
    "\n",
    "#Count Genotypes\n",
    "alleles['popA_hom'] = aA.apply(lambda col: col.str.count(r'1/1|1\\|1')).sum(axis=1)\n",
    "alleles['popA_het'] = aA.apply(lambda col: col.str.count(r'0/1|0\\|1|1/0|1\\|0')).sum(axis=1)\n",
    "alleles['popB_hom'] = aB.apply(lambda col: col.str.count(r'1/1|1\\|1')).sum(axis=1)\n",
    "alleles['popB_het'] = aB.apply(lambda col: col.str.count(r'0/1|0\\|1|1/0|1\\|0')).sum(axis=1)\n",
    "freq = alleles[['CHROM','POS','REF','ALT',\n",
    "                      'f_popA_1','f_popB_1',\n",
    "                      'popA_hom', 'popA_het',\n",
    "                      'popB_hom', 'popB_het',\n",
    "                      'nA','nB']].dropna(axis=\"rows\")\n",
    "\n",
    "#Adjust Frequencies for sites lost/gained (Simulated VCF)\n",
    "if simulated_vcf: \n",
    "    miss_popA = (alleles['popA_0'] == 0) & (alleles['popA_1'] == 0)\n",
    "    alleles.loc[miss_popA, 'f_popA_0'] = 0\n",
    "    alleles.loc[miss_popA, 'f_popA_1'] = 0\n",
    "    miss_popB = (alleles['popB_0'] == 0) & (alleles['popB_1'] == 0)\n",
    "    alleles.loc[miss_popB, 'f_popB_0'] = 0\n",
    "    alleles.loc[miss_popB, 'f_popB_1'] = 0\n",
    "    alleles.loc[miss_popA, 'nA'] = len(popA)\n",
    "    alleles.loc[miss_popA, 'nB'] = len(popB)\n",
    "    freq = alleles[['CHROM','POS','REF','ALT',\n",
    "                      'f_popA_1','f_popB_1',\n",
    "                      'popA_hom', 'popA_het',\n",
    "                      'popB_hom', 'popB_het',\n",
    "                      'nA','nB']]\n",
    "\n",
    "freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "16f2f33b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Define readvcf()\n",
    "def readvcf(vcf_file, pop_file, simulated_vcf=False):\n",
    "    #Import Population File\n",
    "    pops = pd.read_csv(pop_file, sep=\"\\t\", header=None)\n",
    "    popA = pops[pops[0]==\"popA\"][1].astype(str)\n",
    "    popB = pops[pops[0]==\"popB\"][1].astype(str)\n",
    "\n",
    "    #Import VCF\n",
    "    with open(vcf_file) as file:\n",
    "        for line in file:\n",
    "            if line.startswith('#CHROM'):\n",
    "                cols = line.lstrip(\"#\").strip().split(\"\\t\")\n",
    "    data = pd.read_csv(vcf_file, sep='\\t', comment='#', names=cols)\n",
    "\n",
    "    #Extract Genotype Information\n",
    "    loci = data[['CHROM','POS','REF','ALT']]\n",
    "    samples = data.drop(columns=['CHROM', 'POS', 'REF', 'ALT','ID','QUAL','FILTER','INFO','FORMAT'])\n",
    "    samples = samples.apply(lambda col: col.str.split(\":\").str[0])\n",
    "    alleles = pd.concat([loci,samples], axis=1)\n",
    "\n",
    "    #Count alleles\n",
    "    gA = alleles[popA].astype(str)\n",
    "    alleles['popA_0'] = gA.apply(lambda col: col.str.count('0')).sum(axis=1) \n",
    "    alleles['popA_1'] = gA.apply(lambda col: col.str.count('1')).sum(axis=1)\n",
    "    gB = alleles[popB].astype(str)\n",
    "    alleles['popB_0'] = gB.apply(lambda col: col.str.count('0')).sum(axis=1)\n",
    "    alleles['popB_1'] = gB.apply(lambda col: col.str.count('1')).sum(axis=1)\n",
    "           \n",
    "    #Calculate Frequencies (Empirical VCF)\n",
    "    alleles['f_popA_0'] = alleles['popA_0']/(alleles['popA_0'] + alleles['popA_1'])\n",
    "    alleles['f_popA_1'] = alleles['popA_1']/(alleles['popA_0'] + alleles['popA_1'])\n",
    "    alleles['f_popB_0'] = alleles['popB_0']/(alleles['popB_0'] + alleles['popB_1'])\n",
    "    alleles['f_popB_1'] = alleles['popB_1']/(alleles['popB_0'] + alleles['popB_1'])\n",
    "    alleles['nA'] = (alleles['popA_0'] + alleles['popA_1']) / 2\n",
    "    alleles['nB'] = (alleles['popB_0'] + alleles['popB_1']) / 2\n",
    "\n",
    "    #Count Genotypes\n",
    "    alleles['popA_hom'] = aA.apply(lambda col: col.str.count(r'1/1|1\\|1')).sum(axis=1)\n",
    "    alleles['popA_het'] = aA.apply(lambda col: col.str.count(r'0/1|0\\|1|1/0|1\\|0')).sum(axis=1)\n",
    "    alleles['popB_hom'] = aB.apply(lambda col: col.str.count(r'1/1|1\\|1')).sum(axis=1)\n",
    "    alleles['popB_het'] = aB.apply(lambda col: col.str.count(r'0/1|0\\|1|1/0|1\\|0')).sum(axis=1)\n",
    "    freq = alleles[['CHROM','POS','REF','ALT',\n",
    "                      'f_popA_1','f_popB_1',\n",
    "                      'popA_hom', 'popA_het',\n",
    "                      'popB_hom', 'popB_het',\n",
    "                      'nA','nB']].dropna(axis=\"rows\")\n",
    "    freq = freq[(freq[\"nA\"] > 1) & (freq[\"nB\"] > 1)]\n",
    "\n",
    "    #Adjust Frequencies for sites lost/gained (Simulated VCF)\n",
    "    if simulated_vcf: \n",
    "        miss_popA = (alleles['popA_0'] == 0) & (alleles['popA_1'] == 0)\n",
    "        alleles.loc[miss_popA, 'f_popA_0'] = 0\n",
    "        alleles.loc[miss_popA, 'f_popA_1'] = 0\n",
    "        miss_popB = (alleles['popB_0'] == 0) & (alleles['popB_1'] == 0)\n",
    "        alleles.loc[miss_popB, 'f_popB_0'] = 0\n",
    "        alleles.loc[miss_popB, 'f_popB_1'] = 0\n",
    "        alleles.loc[miss_popA, 'nA'] = len(popA)\n",
    "        alleles.loc[miss_popA, 'nB'] = len(popB)\n",
    "        freq = alleles[['CHROM','POS','REF','ALT',\n",
    "                      'f_popA_1','f_popB_1',\n",
    "                      'popA_hom', 'popA_het',\n",
    "                      'popB_hom', 'popB_het',\n",
    "                      'nA','nB']]\n",
    "    return freq\n",
    "\n",
    "#Define importsites()\n",
    "def importsites(neutral_file, mutation_file, derived_sites):\n",
    "    neutral = pd.read_csv(neutral_file, sep='\\t', header=(0))\n",
    "    mutation = pd.read_csv(mutation_file, sep='\\t', header=(0))\n",
    "    neutral.rename(columns={'chromo':'CHROM', 'position':'POS'}, inplace=True)\n",
    "    mutation.rename(columns={'chromo':'CHROM', 'position':'POS'}, inplace=True)\n",
    "    neu_der = pd.merge(derived_sites, neutral, on=['CHROM','POS'],how='inner', indicator=True)\n",
    "    mut_der = pd.merge(derived_sites, mutation, on=['CHROM','POS'],how='inner', indicator=True)\n",
    "    neu_der = neu_der[neu_der['_merge']=='both'].drop(columns=['_merge'])\n",
    "    mut_der = mut_der[mut_der['_merge']=='both'].drop(columns=['_merge'])\n",
    "    #Report Number of Sites\n",
    "    print(\"Number of derived mutations =\", len(mut_der))\n",
    "    return neu_der, mut_der\n",
    "\n",
    "#Define calcRAB()\n",
    "def calcRAB(neu_der, mut_der, seed):\n",
    "    np.random.seed(seed)\n",
    "    index1=np.random.permutation(len(neu_der))[:10000]\n",
    "    neu1=neu_der.iloc[index1]\n",
    "    f_AD = mut_der['f_popA_1']\n",
    "    f_BD = mut_der['f_popB_1']\n",
    "    f_AN = neu1['f_popA_1']\n",
    "    f_BN = neu1['f_popB_1']\n",
    "    LAB = sum(f_AD*(1-f_BD))/sum(f_AN*(1-f_BN))\n",
    "    LBA = sum(f_BD*(1-f_AD))/sum(f_BN*(1-f_AN))\n",
    "    RAB = LAB/LBA\n",
    "    return RAB\n",
    "\n",
    "#Define calcRAB_neu()\n",
    "def calcRAB_neu(neu_der, seed):\n",
    "    np.random.seed(seed)\n",
    "    index1=np.random.permutation(len(neu_der))[:10000]\n",
    "    neu1=neu_der.iloc[index1]\n",
    "    index2=np.random.permutation(len(neu_der))[:10000]\n",
    "    neu2=neu_der.iloc[index2]\n",
    "    f_AD = neu1['f_popA_1']\n",
    "    f_BD = neu1['f_popB_1']\n",
    "    f_AN = neu2['f_popA_1']\n",
    "    f_BN = neu2['f_popB_1']\n",
    "    LAB = sum(f_AD*(1-f_BD))/sum(f_AN*(1-f_BN))\n",
    "    LBA = sum(f_BD*(1-f_AD))/sum(f_BN*(1-f_AN))\n",
    "    RAB_neu = LAB/LBA\n",
    "    return RAB_neu\n",
    "\n",
    "#Define calcRAB_subs()\n",
    "def calcRAB_sub(neu_sub, mut_sub):\n",
    "    index1=np.random.permutation(len(neu_sub))[:10000]\n",
    "    neu1=neu_sub.iloc[index1]\n",
    "    f_AD = mut_sub['f_popA_1']\n",
    "    f_BD = mut_sub['f_popB_1']\n",
    "    f_AN = neu1['f_popA_1']\n",
    "    f_BN = neu1['f_popB_1']\n",
    "    LAB = sum(f_AD*(1-f_BD))/sum(f_AN*(1-f_BN))\n",
    "    LBA = sum(f_BD*(1-f_AD))/sum(f_BN*(1-f_AN))\n",
    "    RAB_sub = LAB/LBA\n",
    "    return RAB_sub\n",
    "\n",
    "#Define samplesites()\n",
    "def samplesites(sites, psites):\n",
    "    nsites = int(round(len(sites) * psites))\n",
    "    indices = np.random.permutation(len(sites))[:nsites]\n",
    "    subsamp = sites.iloc[indices]\n",
    "    return subsamp\n",
    "\n",
    "#Define jackknife()\n",
    "def jackknife(neu_der, mut_der, psites, iter):\n",
    "    if jackknife:\n",
    "        jx = []\n",
    "        for i in range(iter):\n",
    "            neu_sub = samplesites(neu_der, psites)\n",
    "            mut_sub = samplesites(mut_der, psites)\n",
    "            jx.append(calcRAB_sub(neu_sub, mut_sub))\n",
    "    return np.array(jx)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81415909",
   "metadata": {},
   "source": [
    "## Run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "ecaa0988",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of derived mutations = 4\n",
      "PopA Homozygotes =  1\n",
      "PopA Heterozygotes =  7\n",
      "PopB Homozygotes =  1\n",
      "PopB Heterozygotes =  9\n",
      "calcRAB 0.7163863178952442\n",
      "calcRAB_neu 0.9999999999999998\n",
      "jackknife [0.25666064 0.18805467 0.28849752 0.71764261 0.85804196]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([0.19491526, 0.84400202])"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "der = readvcf(\"/Users/abc6435/Desktop/RABvcfs/rab_chr29_test.vcf\", \n",
    "        \"/Users/abc6435/Desktop/RABvcfs/pops.txt\",\n",
    "        simulated_vcf=False)\n",
    "\n",
    "neutral, mutation = importsites('/Users/abc6435/Desktop/RABvcfs/intergenic.txt',\n",
    "                                '/Users/abc6435/Desktop/RABvcfs/tolerated.txt', der)\n",
    "\n",
    "print(\"PopA Homozygotes = \", mutation['popA_hom'].sum())\n",
    "print(\"PopA Heterozygotes = \", mutation['popA_het'].sum())\n",
    "print(\"PopB Homozygotes = \", mutation['popB_hom'].sum())\n",
    "print(\"PopB Heterozygotes = \", mutation['popB_het'].sum())\n",
    "print(\"calcRAB\", calcRAB(neutral, mutation, 20))\n",
    "print(\"calcRAB_neu\",calcRAB_neu(neutral, 20))\n",
    "jx_array=jackknife(neutral, mutation, 0.30, 5)\n",
    "print(\"jackknife\", jx_array)\n",
    "np.percentile(jx_array, [2.5, 97.5])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc8e6842",
   "metadata": {},
   "source": [
    "## Terminal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a921e4c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "calcRAB 0.9821766816874066\n",
      "calcRAB_neu 1.021948329297477\n",
      "jackknife [0.91832788 1.07702961 0.84243057 1.08278136 0.90063503]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([0.84825102, 1.08220619])"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "source ~/RABvcfs_env/bin/activate\n",
    "python3 /Users/abc6435/Desktop/KROH/scripts/analysis/slim/rab/RABvcfs.py --vcf \"rab_chr29_test.vcf\" --pop pops.txt --fileN intergenic.txt --fileM nonsynonymous.txt --seed 34 --psites 0.30 --iter 5 --simulated_vcf"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
